crates/node-optimism/src/node.rs

```diff
@@ -0,0 +1,179 @@
+
+impl OptimismNode {
+    /// Returns a [ComponentsBuilder] configured for a regular Ethereum node.
+    pub fn components<Node>(
+    ) -> ComponentsBuilder<Node, OptimismPoolBuilder, EthereumPayloadBuilder, OptimismNetwork>
+    where
+        Node: FullNodeTypes<Engine = OptimismEngineTypes>,
+    {
+        ComponentsBuilder::default()
+            .node_types::<Node>()
+            .pool(OptimismPoolBuilder::default())
+            .payload(EthereumPayloadBuilder::default())
+            .network(OptimismNetwork)
+    }
+}
+
+impl NodeTypes for OptimismNode {
+    type Primitives = ();
+    type Engine = OptimismEngineTypes;
+    type Evm = OptimismEvmConfig;
+
+    fn evm_config(&self) -> Self::Evm {
+        todo!()
+    }
+}
+
+/// A basic ethereum transaction pool.
+///
+/// This contains various settings that can be configured and take precedence over the node's
+/// config.
+#[derive(Debug, Default, Clone, Copy)]
+#[non_exhaustive]
+pub struct OptimismPoolBuilder {
+    // TODO add options for txpool args
+}
+
+impl<Node> PoolBuilder<Node> for OptimismPoolBuilder
+where
+    Node: FullNodeTypes,
+{
+    type Pool = EthTransactionPool<Node::Provider, DiskFileBlobStore>;
+
+    fn build_pool(self, ctx: &BuilderContext<Node>) -> eyre::Result<Self::Pool> {
+        let data_dir = ctx.data_dir();
+        let blob_store = DiskFileBlobStore::open(data_dir.blobstore_path(), Default::default())?;
+        let validator = TransactionValidationTaskExecutor::eth_builder(ctx.chain_spec())
+            .with_head_timestamp(ctx.head().timestamp)
+            .kzg_settings(ctx.kzg_settings()?)
+            .with_additional_tasks(1)
+            .build_with_tasks(
+                ctx.provider().clone(),
+                ctx.task_executor().clone(),
+                blob_store.clone(),
+            );
+
+        let transaction_pool =
+            reth_transaction_pool::Pool::eth_pool(validator, blob_store, ctx.pool_config());
+        info!(target: "reth::cli", "Transaction pool initialized");
+        let transactions_path = data_dir.txpool_transactions_path();
+
+        // spawn txpool maintenance task
+        {
+            let pool = transaction_pool.clone();
+            let chain_events = ctx.provider().canonical_state_stream();
+            let client = ctx.provider().clone();
+            let transactions_backup_config =
+                reth_transaction_pool::maintain::LocalTransactionBackupConfig::with_local_txs_backup(transactions_path);
+
+            ctx.task_executor().spawn_critical_with_graceful_shutdown_signal(
+                "local transactions backup task",
+                |shutdown| {
+                    reth_transaction_pool::maintain::backup_local_transactions_task(
+                        shutdown,
+                        pool.clone(),
+                        transactions_backup_config,
+                    )
+                },
+            );
+
+            // spawn the maintenance task
```

Comment:
maintenance gives the notion it's a background task that is periodically doing some optimisation task. for for example reordering elements in some store or moving elements between two different stores.